helix-term/src/application.rs

```diff
@@ -228,6 +229,63 @@ impl Application {
                     Notification::LogMessage(params) => {
                         log::warn!("unhandled window/logMessage: {:?}", params);
                     }
+                    Notification::ProgressMessage(params) => {
+                        let token = match params.token {
+                            lsp::NumberOrString::Number(n) => n.to_string(),
```

Comment:
One more thing before I merge. Can we reuse the `String` buffer across notifications?

Reply:
I was wondering if we could simplify some of the messages using `format!()` too, maybe using some kind of match statement (one that maps to `(title, message, percentage)`, then another one matching that `(Some(..), None, None)` for example and using `format!`

Reply:
Updated to use `match` with `format!`